sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -46,9 +46,13 @@ class QueryExecution(val sparkSession: SparkSession, val logical: LogicalPlan) {
   protected def planner = sparkSession.sessionState.planner
 
   def assertAnalyzed(): Unit = {
-    try sparkSession.sessionState.analyzer.checkAnalysis(analyzed) catch {
+    var analyzedPlan: Option[LogicalPlan] = None
```

Comment:
You could make this a local lazy val. That should be a bit more concise. For example:
```scala
def assertAnalyzed(): Unit = {
  lazy val analyzedPlan = analyzed
  try {
    sparkSession.sessionState.analyzer.checkAnalysis(analyzedPlan.get)
  } catch {
    case e: AnalysisException =>
      val ae = new AnalysisException(e.message, e.line, e.startPosition, Option(analyzedPlan))
      ae.setStackTrace(e.getStackTrace)
      throw ae
  }
}
```